### PR TITLE
Solutions to Chapter One

### PR DESCRIPTION
**Pull request by mstruebing**
### Solutions for Chapter {NUMBER}

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -473,7 +473,8 @@ Implement a function that returns the last digit of a given number.
   whether it works for you!
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = mod n 10
```
Comment (vrom911) [inline src/Chapter1.hs]: Almost there! Unfortunately, it won't work properly on negative numbers. You would need to use one more standard function 🙂

File: src/Chapter1.hs [hunk 2]
```diff
@@ -537,7 +538,10 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z = head $ filter (/= minimum xs) $ filter (/= maximum xs) xs
+    where
+        xs = [x, y, z]
```
Comment (vrom911) [inline src/Chapter1.hs]: Wow! That is a clever solution 🧠 and very creative!

Going deeper into your idea, you could also do something like this:
```haskell
mid x y z = let (_:y:_) = sort [x, y, z] in y
```

File: src/Chapter1.hs [hunk 3]
```diff
@@ -551,7 +555,14 @@ True
...
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+isVowel :: Char -> Bool
+isVowel c 
+  | c == 'A' = True
+  | c == 'E' = True
+  | c == 'I' = True
+  | c == 'O' = True
+  | c == 'U' = True
```
Comment (vrom911) [inline src/Chapter1.hs]: Oh, that's actually a good point that uppercased vowels are vowels as well. But for this task, we thought about checking only lowercase ones 😄

File: src/Chapter1.hs [hunk 4]
```diff
@@ -614,8 +625,10 @@ Implement a function that returns the sum of the last two digits of a number.
 Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
-
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = lastDigit n + secondLast
```
Comment (vrom911) [inline src/Chapter1.hs:632]: That's cool! Reusage of already written functions is smart! 🔝

File: src/Chapter1.hs [hunk 5]
```diff
@@ -614,8 +625,10 @@ Implement a function that returns the sum of the last two digits of a number.
...
+    where
+        secondLast = lastDigit $ div n 10
```
Comment (vrom911) [inline src/Chapter1.hs]: I think you would need to take `abs n` to work properly on negatives 🙂

File: src/Chapter1.hs [hunk 6]
```diff
@@ -636,7 +649,10 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n 
+    | div n 10 /= 0 = firstDigit $ div n 10
+    | otherwise = n
```
Comment (vrom911) [inline src/Chapter1.hs:658]: In the case of `-1` it would return `-1` thought it should be `1` 🙂 I would suggest to also look at the negative case 👍🏼

**Review by vrom911 — APPROVED** [review]
Great start! 🌟 

I have noticed that you skipped some of the tasks, do you need help with GHCI tasks? Or that is okay? 🙂 We are here to help, if you have any questions!

**Comment by chshersh** [comment]
⚠️ By the way, per new rules of the Hacktoberfest event, you would need to add the `hacktoberfest` topic to this fork repo:

* https://hacktoberfest.digitalocean.com/hacktoberfest-update

All previous contributions are legit, but from now on it should have the label to participate.

**Pull request merged**